cdap-api/src/main/java/co/cask/cdap/api/dataset/DatasetDefinition.java

```diff
@@ -71,24 +74,28 @@
    * Provides dataset admin to be used to perform administrative operations on the dataset instance defined by passed
    * {@link DatasetSpecification}.
    *
-   * @param spec specification of the dataset instance.
+   *
+   * @param datasetContext context for the dataset
    * @param classLoader classloader to use when executing admin operations
+   * @param spec specification of the dataset instance.
    * @return dataset admin to perform administrative operations
    * @throws IOException
    */
-  A getAdmin(DatasetSpecification spec, ClassLoader classLoader) throws IOException;
+  A getAdmin(DatasetContext datasetContext, ClassLoader classLoader, DatasetSpecification spec) throws IOException;
```

Comment:
Correct me if I am wrong. The ``DatasetAdminOpHttpHandler`` is the one that will call these methods, right? It will get a request for a dataset with a namespace-id in the request. It uses that namespace-id to create the ``DatasetContext`` and calls this method. If the namespace id is wrong (or you use a namespace that you do not have access to), the admin operation will fail.

Anything that I am missing?